chpl/chpl-api/src/main/java/gov/healthit/chpl/web/controller/StatusController.java

```diff
@@ -42,24 +45,27 @@
      * @return JSON object with status of load and "age"
      */
     @ApiOperation(
-            value = "Check if the basic search cache has completed loading. "
-                    + "{ status: 'OK', age: long } is returned if it's finished and "
-                    + "{ status: 'INITIALIZING' } is returned if not. "
-                    + "Age indicates the number of miliseconds since the cache "
-                    + "was last refreshed.",
+            value = "Check the status of every cache. "
+                    + "{ status: 'OK' } is returned if all caches are loaded and "
+                    + "{ status: 'PENDING' } is returned if not. ",
                     notes = "")
     @RequestMapping(value = "/cache_status", method = RequestMethod.GET, produces = "application/json; charset=utf-8")
-    public @ResponseBody String getCacheStatus() {
+    public @ResponseBody CacheStatus getCacheStatus() {
+        CacheStatus response = new CacheStatus();
         CacheManager manager = CacheManager.getInstance();
-        Cache basicCache = manager.getCache(CacheNames.COLLECTIONS_LISTINGS);
-        if (basicCache == null || basicCache.getKeysNoDuplicateCheck().size() == 0) {
-            return "{\"status\": \"" + CacheStatus.INITIALIZING + "\"}";
+        boolean anyPending = false;
+        List<String> cacheNames = CacheInitializor.getPreInitializedCaches();
+        for(int i = 0; i < cacheNames.size(); i++) {
```

Comment:
I know it's minor, but checkstyle complains about `if(` instead of `if (` and similarly with for/try/catch/etc.